src/support/socket.h

```diff
@@ -85,7 +85,7 @@ inline std::string GetHostName() {
  * \return result of operation.
  */
 inline bool ValidateIP(std::string ip) {
-  if (ip == "localhost") {
+  if (ip == "127.0.0.1") {
```

Comment:
This should not be necessary since "127.0.0.1" should pass the is_ipv4 check below.

Reply:
This should have been left as it was with "localhost" since it is a special valid address.